OpenEdXMobile/src/main/java/org/edx/mobile/module/analytics/InAppPurchasesAnalytics.kt

```diff
@@ -0,0 +1,156 @@
+package org.edx.mobile.module.analytics
+
+import org.edx.mobile.core.IEdxEnvironment
+import java.util.*
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class InAppPurchasesAnalytics @Inject constructor(
+    var environment: IEdxEnvironment
+) {
+    private var courseId: String = ""
+    private var isSelfPaced: Boolean = false
+    private var screenName: String = ""
+    private var componentId: String = ""
+    private var price: String = ""
+    private var elapsedTime: Long = 0
+    private var loadPriceTime: Long = 0
+    private var paymentTime: Long = 0
+    private var unlockContentTime: Long = 0
+    private var refreshContentTime: Long = 0
+    private var upgradeCourseTime: Long = 0
+    private var errorMsg: String = ""
+    private var errorAction: String = ""
+
+    fun initCourseValues(
+        courseId: String,
+        isSelfPaced: Boolean,
+        screenName: String,
+        componentId: String = ""
+    ) {
+        this.courseId = courseId
+        this.isSelfPaced = isSelfPaced
+        this.screenName = screenName
+        this.componentId = componentId
+        this.price = ""
+        resetAnalytics()
+    }
+
+    fun initPriceTime() {
+        loadPriceTime = getCurrentTime()
+    }
+
+    private fun initCourseUpgradeTime() {
+        upgradeCourseTime = getCurrentTime()
+    }
+
+    fun initUnlockContentTime() {
+        unlockContentTime = getCurrentTime()
+    }
+
+    fun initPaymentTime() {
+        paymentTime = getCurrentTime()
+    }
+
+    fun initRefreshContentTime() {
+        refreshContentTime = getCurrentTime()
+    }
+
+    fun setPrice(price: String) {
+        this.price = price
+    }
+
+    fun trackIAPEvent(
+        eventName: String,
+        errorMsg: String = "",
+        actionTaken: String = ""
+    ) {
+        when (eventName) {
+            Analytics.Events.IAP_UPGRADE_NOW_CLICKED -> {
+                trackEvent(eventName, Analytics.Values.IAP_UPGRADE_NOW_CLICKED)
+                initCourseUpgradeTime()
```

Comment:
better to assign the value to property `upgradeCourseTime`. cuz only using within the class